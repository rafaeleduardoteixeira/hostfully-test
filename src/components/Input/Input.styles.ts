import styled from 'styled-components';
import { colors, media } from '../../themes';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

export const InputLabel = styled.label`
  font-size: 1rem;
  font-weight: 500;
  margin-bottom: 5px;
  color: ${colors.grayDark};
`;

export const InputField = styled.input<{ width?: string; error?: string }>`
  height: 50px;
  border: none;
  outline: none;
  padding: 0 10px 0 20px;
  font-size: 1rem;
  color: ${colors.black};
  background-color: ${colors.grayInput};
  border-radius: 10px;
  width: ${(props) => props.width || '90%'};

  outline: ${(props) => (props.error ? `1px solid ${colors.red}` : 'none')};

  &::placeholder {
    color: ${colors.gray};
  }

  &:focus {
    outline: ${colors.grayLight} auto 1px;
  }

  &:disabled {
    cursor: not-allowed;
  }

  @media (${media.mobile}) {
    font-size: 0.9rem;
    height: 40px;
  }
`;
