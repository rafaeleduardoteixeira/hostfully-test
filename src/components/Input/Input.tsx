import React, { ChangeEvent } from 'react';
import { Container, InputLabel, InputField } from './Input.styles';

interface InputProps {
  name?: string;
  label?: string;
  value?: string;
  type?: string;
  placeholder?: string;
  width?: string;
  disabled?: boolean;
  error?: string;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
}

export const Input = (props: InputProps) => {
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    props.onChange && props.onChange(event);
  };

  return (
    <Container>
      {props.label && <InputLabel htmlFor={props.name}>{props.label}</InputLabel>}
      <InputField data-testid="Input" {...props} onChange={handleChange} />
    </Container>
  );
};
