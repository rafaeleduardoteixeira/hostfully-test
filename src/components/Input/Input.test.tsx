import React, { act } from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Input } from './Input';

describe('Input', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render Input', () => {
    render(<Input />);
    const InputElement = screen.getByTestId('Input');
    expect(InputElement).toBeInTheDocument();
  });
  test('Should render Input with label', () => {
    render(<Input label="Label" />);
    const InputElement = screen.getByTestId('Input');
    const LabelElement = screen.getByText('Label');
    expect(InputElement).toBeInTheDocument();
    expect(LabelElement).toBeInTheDocument();
  });
  test('Should type in Input', () => {
    render(<Input onChange={jest.fn()} />);
    const InputElement = screen.getByTestId('Input');
    act(() => {
      fireEvent.change(InputElement, { target: { value: 'a' } });
    });
    expect(InputElement).toHaveValue('a');
  });
});
