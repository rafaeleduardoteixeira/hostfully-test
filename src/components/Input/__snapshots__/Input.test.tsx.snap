// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu jqDACe"
  >
    <input
      class="sc-elDIKY hDaMyU"
      data-testid="Input"
    />
  </div>
</DocumentFragment>
`;
